spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/repository/query/PartTreeFirestoreQuery.java

```diff
@@ -86,14 +90,26 @@ public Object execute(Object[] parameters) {
 		}
 
 		StructuredQuery.Builder builder = createBuilderWithFilter(parameters);
-		Flux<?> result = this.reactiveOperations.execute(builder, this.persistentEntity.getType());
 
+		// Handle Pageable parameters.
+		if (!getQueryMethod().getParameters().isEmpty()) {
```

Comment:
Can you just call `getQueryMethod().isPageQuery()` here?

Reply:
No, the getQueryMethod().isPageQuery() determines if its a page query if the return type is a subclass of Page.

```
  public final boolean isPageQuery() {
    return org.springframework.util.ClassUtils.isAssignable(Page.class, this.unwrappedReturnType);
  }
```

However we don't want to deal with Page or Slice objects because they are not reactive; they store the contents of the query into a list and allow you to manipulate them further.

See Mark's comments here: https://stackoverflow.com/a/46391830/906497

So in this case we are modelling our behavior after Mongo reactive repository's pattern of allowing LIMIT and OFFSET and then transmitting the results in a flux.

Reply:
@dzou could you clarify why sorting is blocking? I was unable to find that in the linked page.

Reply:
Yup, sorting is a blocking operation because one must wait for all the elements in the flux to arrive (wait for flux to complete) before performing sort and then returning the result to the user.

Reactive lets you sort the elements of a flux; you can see what I mean - Its return type is `Mono<List<T>>` not `Flux<T>`:
https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#collectSortedList--

Reply:
We are not going to sort locally, we would just send sorting options to firestore, where everything is indexed and can be efficiently streamed to us in the requested order.

Reply:
Ohh I see, that makes sense.